homeassistant/components/ted5000/sensor.py

```diff
@@ -29,16 +34,127 @@
 
+SENSOR_TYPES_BASIC: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ENTITY_MTU_POWER,
+        name="Power",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=POWER_WATT,
+    ),
+    SensorEntityDescription(
+        key=ENTITY_MTU_VOLTAGE,
+        name="Voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+    ),
+)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+SENSOR_TYPES_ADVANCED: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key=ENTITY_MTU_TODAY,
+        name="Energy Daily",
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SensorEntityDescription(
+        key=ENTITY_MTU_MONTH,
+        name="Energy Monthly",
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    SensorEntityDescription(
+        key=ENTITY_MTU_PF,
+        name="Power Factor",
+        device_class=SensorDeviceClass.POWER_FACTOR,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+)
+
+SENSOR_TYPES_EXTENDED: tuple[SensorEntityDescription, ...] = (
+    # MTUs Quantity
+    SensorEntityDescription(
+        key=ENTITY_MTUS,
+        name="MTU Quantity",
+    ),
+    # Current Rate $/kWh
+    SensorEntityDescription(
+        key=ENTITY_RATE,
+        name="Current Rate",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="$/kWh",
+    ),
+    # Days left in billing cycle
+    SensorEntityDescription(
+        key=ENTITY_DAYSLEFT,
+        name="Days Left Billing Cycle",
+        native_unit_of_measurement=TIME_DAYS,
+    ),
+    # Plan type (Flat, Tier, TOU, Tier+TOU)
+    SensorEntityDescription(
+        key=ENTITY_PLANTYPE,
+        name="Plan Type",
+    ),
+    # Current Tier (0 = Disabled)
+    SensorEntityDescription(
+        key=ENTITY_TIER,
+        name="Current Tier",
+    ),
+    # Current TOU (0 = Disabled)
+    SensorEntityDescription(key=ENTITY_TOU, name="Current TOU"),
+    # Current TOU Description (if Current TOU is 0 => Not Configured)
+    SensorEntityDescription(
+        key=ENTITY_TOUDESC,
+        name="Current TOU Description",
+    ),
+    # Carbon Rate lbs/kW
+    SensorEntityDescription(
+        key=ENTITY_CARBONRATE,
+        name="Carbon Rate",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement="lbs/kW",
+    ),
+    # Meter read date
+    SensorEntityDescription(
```

Comment:
No, it can't be a DATE class as it only returns an integer of the reading day, e.g. "1" for a reading on the first of the month
![image](https://user-images.githubusercontent.com/13887820/196193173-e74b0ec9-6d9c-4662-a54b-9a813c7eb5f5.png)

Reply:
In that case the entity name might be better as 'Day of Month'

Reply:
This is the exact name given in the TED API and on the web browser